apps/analysis/steps/parsers.py

```diff
@@ -0,0 +1,45 @@
+import re
+from datetime import datetime
+
+import pandas as pd
+
+from apps.analysis.core import BaseStep, Params
+from apps.analysis.exceptions import StepError
+
+
+class WhatsappParser(BaseStep[str, pd.DataFrame]):
+    """Parse a whatsapp chat log into a dataframe.
+
+    Assumes the following format:
+
+        01/01/2021, 00:00 - System Message
+        01/01/2021, 00:00 - Sender: Message which can
+        go across multiple lines
+    """
+
+    input_type = str
+    output_type = pd.DataFrame
+
+    def run(self, params: Params, data: str) -> tuple[pd.DataFrame, dict]:
```

Comment:
Have you had a look at the [whatstk](https://pypi.org/project/whatstk/) library?

Reply:
Yes I did look at that but it seemed way overkill for just the parsing. They also take quite a complex approach to parsing which didn't seem necessary (unless there are multiple chat formats - I'm assuming there's only one).